statsmodels/stats/pscore_match.py

```diff
@@ -0,0 +1,242 @@
+        min_control, max_control = self.scores[control].min(), self.scores[control].max()
+        common_min, common_max = max(min_treated, min_control), min(max_treated, max_control)
+        return (self.scores >= common_min) & (self.scores <= common_max)
+        
+        
+class PScoreMatchResult(object):
+    pass
+
+class Stratum(object):
+    def __init__(self, ps, min, max):
+        self.ps = ps
+        self.index = ps.common_support() & (ps.scores >= min) & (ps.scores < max)
+        self.min, self.max = min, max
+        
+    def bisect(self):
+        half = (self.min + self.max)/2
+        return Stratum(self.ps, self.min, half), Stratum(self.ps, half, self.max)
+        
+    def describe(self):
+        scores = self.ps.scores[self.index]
+        min, max = scores.min(), scores.max()
+        half = (min+ max)/2
+        print 'stratum min, half, max: ' + str([min, half, max])
+        print 'Treated, Control:' + str([self.ps.assigment_index[x].count() for x in (self.treated(), self.control())])
+        
+    def treated(self):
+        return self.ps.treated() & self.index 
+        
+    def control(self):
+        return self.ps.control() & self.index
+        
+    def check_balance(self):
+        return self.check_balance_for(self.ps.scores) and self.check_balance_for(self.ps.covariates)
+        
+    def check_balance_for(self, data):
+        treated, control = data[self.treated()], data[self.control()]
+        cm = CompareMeans(DescrStatsW(treated), DescrStatsW(control))
+        test = cm.ttest_ind()
+        pvalues = test[1]
+        alpha = 0.005
+        return np.all(pvalues > alpha)
+        
+    def treatment_effect(self):
+        objective = self.ps.treatment_objective_variable
+        diff = objective[self.treated()].mean() - objective[self.control()].mean()
+        return diff
+    
+    def weighted_treatment_effeect(self):
+        return self.treatment_effect() * self.weight()
+    
+    def weight(self):
+        return self.ps.assigment_index[self.treated()].count()
+        
+    def empty(self):
+        return not (np.any(self.treated()) & np.any(self.control()))
+
+class StrataMatchingAlgorithm(object):
+    def __init__(self, psmatch):
+        self.ps = psmatch
+        
+    def compute_strata(self, strat):
+        if strat.empty():
+            return []
+        if strat.check_balance():
+            return [strat,]
+        else:
+            left, right = strat.bisect()
+            return self.compute_strata(left) + self.compute_strata(right)
+            
+            
+    def basic_strata(self):
+        #limits = np.linspace(scores[common].min(), scores[common].max(), 5+1)
+        limits = np.linspace(0, 1, 5+1)
+        strata = []
+        min = limits[0]
+        for max in limits[1:]:
+            strat = Stratum(self.ps, min, max)
+            strata.append(strat)
+            min = max
+        return strata
+        
+    def fit(self):
```

Comment:
Might want a different name for this method since we usually use 'fit' to fit models to data.